### PR TITLE
Add WebP support in site health

### LINKED ISSUE #130: Expose WebP support under Site Health status tab
Since not every server environment supports translating JPEG uploads to WebP format, it's worth indicating this to site administrators in an easy-to-understand way. While we already have some basic PHP extension information about image capabilities in the Site Health info tab, it may be worth adding something explicitly about WebP in the Site Health "Status" tab.

For example a message like "WebP images are (not) supported by your server environment", with a brief description on why this is relevant. And potentially a note to "Ask your hosting provider to enable/update xyz"

Also see https://wordpress.slack.com/archives/C02KGN5K076/p1643112555307200?thread_ts=1643041077.295600&cid=C02KGN5K076

cc @adamsilverstein @getsource @spacedmonkey

### PR DESCRIPTION
**Pull request by kirtangajjar**
## Summary
Fixes #130. 

## Relevant technical choices

Adds a module in site health to check for WebP support and shows a warning if support is not present. 
If either GD or Imagick has WebP support, the WebP support warning is not shown.

The code of checking support in GD and Imagick is same as how it's shown in site health info's media handling section.

## Screenshots
Screenshot from 2022-02-03 18-19-09

### CODE REVIEW

**Comment by pbearne** [comment]
it's nice to see the output But most users won't know what to look for
I feel we need to add a "score" as to how good the support is. 
ie a nice green 100% support if all is good and messages, if support expected/desired, is missing

**Comment by kirtangajjar** [comment]
@pbearne Can you look at the screenshot again? Earlier I added a screenshot in the last line to indicate where I copied the code of checking WebP support from, but I saw how it can be mistaken as the output of the code, so I have uploaded the actual screenshot of output. 

Do you still feel we need to show percentage-wise support? I think now since the user has a clear picture to see that they have an action to take here, I feel we don't need to show partial support indication.

File: modules/site-health/webp-uploads-enabled/load.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Callback for webp_enabled test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function webp_uploads_check_webp_supported_test() {
+	$result = array(
+		'label'       => __( 'Your site supports WebP' ),
```
Comment (mitogh) [inline modules/site-health/webp-uploads-enabled/load.php]: Missing the domain name on the translation.

File: modules/site-health/webp-uploads-enabled/load.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance' ),
```
Comment (mitogh) [inline modules/site-health/webp-uploads-enabled/load.php]: Same as above, missing translation domain.

File: modules/site-health/webp-uploads-enabled/load.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			__( 'WebP image format is used by WordPress to boost performance of your site by generating smaller images than it usually could with default JPEG format. This means your pages will load faster and consume less data from users.' )
```
Comment (mitogh) [inline modules/site-health/webp-uploads-enabled/load.php]: Missing domain as well. Some suggestions for the wording of this section.

```suggestion
			__( 'WebP image format is used by WordPress to improve the performance of your site by generating smaller images than it usually could with JPEG format. This means your pages will load faster and consume less bandwidth from users.' )
```

File: modules/site-health/webp-uploads-enabled/load.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	}
+
+	try {
+		$formats = Imagick::queryFormats( '*' );
+
+		if( in_array('WEBP', $formats ) ) {
+			$gd_supported = true;
+		}
+	} catch (Exception $e) {
+	}
```
Comment (mitogh) [inline modules/site-health/webp-uploads-enabled/load.php]: Looks like this section of code can be replaced with: 

```php
wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) 
```

As that would use the supported image editor if Imagemagic is supported or GD if that's the case.

**Review by mitogh — CHANGES_REQUESTED** [review]

File: modules/site-health/webp-uploads-enabled/load.php (hunk 1 above)
Comment (kirtangajjar) [reply modules/site-health/webp-uploads-enabled/load.php]: Fixed

File: modules/site-health/webp-uploads-enabled/load.php (hunk 2 above)
Comment (kirtangajjar) [reply modules/site-health/webp-uploads-enabled/load.php]: Fixed

File: modules/site-health/webp-uploads-enabled/load.php (hunk 4 above)
Comment (kirtangajjar) [reply modules/site-health/webp-uploads-enabled/load.php]: Thanks for the tip. I've made the change and the code looks much cleaner :)

**Review by mitogh — COMMENTED** [review]
This looks good to me, let's wait on other people to chime in case they have any additional feedback.

Thank you for putting this together :1st_place_medal: :bow:

**Comment by mmuyskens** [comment]
I wouldn't call it critical. Kick it down to recommendations.

File: modules/site-health/webp-support/load.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+	$webp_supported = wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) );
+
+	if ( ! $webp_supported ) {
+		$result['status']  = 'critical';
+		$result['label']   = __( 'Your site does not support WebP', 'performance-lab' );
+		$result['actions'] = sprintf(
+			'<p>%s</p>',
+			/* translators: Accessibility text. */
+			__( 'Please contact your host and ask them to add WebP support', 'performance-lab' )
```
Comment (adamsilverstein) [inline modules/site-health/webp-support/load.php]: Since this is a full sentence, it should end in a period.

**Comment by adamsilverstein** [comment]
Nice work @kirtangajjar - code looks good, I'll give this a test.

**Comment by adamsilverstein** [comment]
This worked well in my testing! 

After testing though, I feel like the notice belongs under "recommended improvements" instead of "critical issues". Having WebP support is nice to have, it isn't really critical.

File: modules/site-health/webp-support/load.php [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+			__( 'WebP image format is used by WordPress to improve the performance of your site by generating smaller images than it usually could with JPEG format. This means your pages will load faster and consume less bandwidth from users.', 'performance-lab' )
+		),
+		'actions'     => '',
+		'test'        => 'is_webp_uploads_enabled',
+	);
+
+	$webp_supported = wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) );
+
+	if ( ! $webp_supported ) {
+		$result['status']  = 'critical';
```
Comment (adamsilverstein) [inline modules/site-health/webp-support/load.php]: Suggest changing 'critical' to 'recommended'
```suggestion
		$result['status']  = 'recommended';
```

**Comment by kirtangajjar** [comment]
@adamsilverstein Done. LMK if you find any additional changes.

**Review by mitogh — APPROVED** [review]

File: modules/site-health/webp-support/load.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	$result = array(
+		'label'       => __( 'Your site supports WebP', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			__( 'WebP image format is used by WordPress to improve the performance of your site by generating smaller images than it usually could with JPEG format. This means your pages will load faster and consume less bandwidth from users.', 'performance-lab' )
```
Comment (adamsilverstein) [inline modules/site-health/webp-support/load.php]:
```suggestion
			__( 'WebP image format is used by WordPress to improve the performance of your site by generating smaller images than it usually could with the JPEG format. This means your pages will load faster and consume less bandwidth.', 'performance-lab' )
```
Slight text edit suggestion

**Comment by adamsilverstein** [comment]
@kirtangajjar I left one additional tiny suggestion on the text, then we can merge this. Thanks for your work here!

**Comment by kirtangajjar** [comment]
@adamsilverstein I've committed your suggestion. Thanks for giving it a second look!

**Comment by adamsilverstein** [comment]
👍🏼  Looks good, nice work @kirtangajjar - going to go ahead and merge this. We'll get another chance to review the exact wording when we propose this feature for core.

**Review by adamsilverstein — APPROVED** [review]

**Pull request merged**